### PR TITLE
Added PID pybind11 interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

This PR creates a PID Pybind11 interface

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/python_pybind11/test/PID_TEST.py [hunk 1]
```diff
@@ -181,9 +183,7 @@ def test_zero_gains(self):
 
         self.update_test(pid, 0,  1, -1,  1, 0, 0)
         self.update_test(pid, 0,  1, -1,  1, 0, 0)
-        self.update_test(pid, 0, -1, -1, -1, 0, 2)
         self.update_test(pid, 0, -1, -1, -1, 0, 0)
-        self.update_test(pid, 0,  1, -1,  1, 0, -2)
```
Comment (chapulina) [inline src/python_pybind11/test/PID_TEST.py:186]: Why are these 2 test cases being removed?

**Review by chapulina — COMMENTED** [review]
Just a question about tests

File: src/python_pybind11/test/PID_TEST.py (hunk 1 above)
Comment (ahcorde) [reply src/python_pybind11/test/PID_TEST.py:186]: I didn't dig a lot on this, but could be related with the python module datetime timedelta

Comment (chapulina) [reply src/python_pybind11/test/PID_TEST.py:186]: I added some TODOs in 980b23b7347dbe2c0cfcd09caa34cbe242525a1c, we should look into this after the pybind11 migration is over. It's strange that the results are different from SWIG and C++

**Review by chapulina — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**